pkg/console/controllers/resourcesyncdestination/controller.go

```diff
@@ -27,118 +26,103 @@ const (
-	defer runtime.HandleCrash()
-	defer c.queue.ShutDown()
-	klog.Infof("starting %v", controllerName)
-	defer klog.Infof("shutting down %v", controllerName)
-	if !cache.WaitForCacheSync(stopCh, c.cachesToSync...) {
-		klog.Infoln("caches did not sync")
-		runtime.HandleError(fmt.Errorf("caches did not sync"))
-		return
+	ingressCertErr := c.syncIngressCert(ctx)
+	statusHandler.AddCondition(status.HandleDegraded("IngressCertSync", "FailedIngressCertSync", ingressCertErr))
+	if ingressCertErr != nil {
+		return statusHandler.FlushAndReturn(ingressCertErr)
 	}
-	// only start one worker
-	go wait.Until(c.runWorker, time.Second, stopCh)
-	<-stopCh
-}
 
-func (c *ResourceSyncDestinationController) runWorker() {
-	for c.processNextWorkItem() {
+	customLogoErr := c.syncCustomLogo(ctx, updatedOperatorConfig)
+	statusHandler.AddCondition(status.HandleDegraded("CustomLogoSync", "FailedCustomLogoSync", customLogoErr))
+	if customLogoErr != nil {
+		return statusHandler.FlushAndReturn(customLogoErr)
 	}
+
+	return err
 }
 
-func (c *ResourceSyncDestinationController) processNextWorkItem() bool {
-	processKey, quit := c.queue.Get()
-	if quit {
-		return false
-	}
-	defer c.queue.Done(processKey)
-	err := c.sync()
-	if err == nil {
-		c.queue.Forget(processKey)
-		return true
+func (c *ResourceSyncDestinationController) syncCustomLogo(ctx context.Context, operatorConfig *operatorsv1.Console) error {
+	source := resourcesynccontroller.ResourceLocation{}
+	logoConfigMapName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+
+	if logoConfigMapName != "" {
+		source.Name = logoConfigMapName
+		source.Namespace = api.OpenShiftConfigNamespace
 	}
-	runtime.HandleError(fmt.Errorf("%v failed with : %v", processKey, err))
-	c.queue.AddRateLimited(processKey)
-	return true
+	// if no custom logo provided, sync an empty source to delete
+	return c.resourceSyncer.SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{Namespace: api.OpenShiftConsoleNamespace, Name: api.OpenShiftCustomLogoConfigMapName},
+		source,
+	)
 }
 
-func (c *ResourceSyncDestinationController) newEventHandler() cache.ResourceEventHandler {
-	return cache.ResourceEventHandlerFuncs{
-		AddFunc:    func(obj interface{}) { c.queue.Add(controllerWorkQueueKey) },
-		UpdateFunc: func(old, new interface{}) { c.queue.Add(controllerWorkQueueKey) },
-		DeleteFunc: func(obj interface{}) { c.queue.Add(controllerWorkQueueKey) },
-	}
+func (c *ResourceSyncDestinationController) syncIngressCert(ctx context.Context) error {
+	// sync: 'default-ingress-cert' configmap
+	// from: 'openshift-config-managed' namespace
+	// to:   'openshift-console' namespace
+	return c.resourceSyncer.SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{Name: api.DefaultIngressCertConfigMapName, Namespace: api.OpenShiftConsoleNamespace},
+		resourcesynccontroller.ResourceLocation{Name: api.DefaultIngressCertConfigMapName, Namespace: api.OpenShiftConfigManagedNamespace},
+	)
```

Comment:
These appear to be static, you can just move them out to the starter, just like 
https://github.com/openshift/cluster-authentication-operator/blob/cf1e1a6c79db7cf29e7de2a90ecdc458bc13059c/pkg/operator/starter.go#L198-L204

```go
	// add syncing for the OAuth metadata ConfigMap
	if err := operatorCtx.resourceSyncController.SyncConfigMap(
		resourcesynccontroller.ResourceLocation{Namespace: "openshift-config-managed", Name: "oauth-openshift"},
		resourcesynccontroller.ResourceLocation{Namespace: "openshift-authentication", Name: "v4-0-config-system-metadata"},
	); err != nil {
		return err
	}
```